### ISSUE TITLE
🐛 Bug: rule `restrict-dependency-ranges` don't output any error

### ISSUE THREAD

**Issue by gadzhimari** [JoshuaKGoldberg/eslint-plugin-package-json#1893]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have [searched for related issues](https://github.com/michaelfaith/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Show error that should be triggered by `restrict-dependency-ranges` rule. 

### Actual

I don't see any error in package.json related to ESLint

<img width="2332" height="884" alt="Image" src="https://github.com/user-attachments/assets/88302f16-40fc-440c-8d17-081e4bdd94b1" />

### Version

1.2.0

### ESLint Version

8.48.0

### Additional Info

In my project I have the following .eslintrc file. I tried everything that described in documentation, but still not helped me. Here my config:

```
{
  "parser": "@typescript-eslint/parser", // Specifies the ESLint parser
  "extends": [
    "plugin:import/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:react/jsx-runtime",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "plugin:prettier/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "env": {
    "browser": true,
    "es2021": true
  },
  "parserOptions": {
    "ecmaVersion": "latest", // Allows for the parsing of modern ECMAScript features
    "sourceType": "module", // Allows for the use of imports
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": ["react", "prettier", "package-json"],
  "rules": {
    // Place to specify ESLint rules. Can be used to overwrite rules
    // specified from the extended configs
    // e.g. "@typescript-eslint/explicit-function-return-type": "off",
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "import/prefer-default-export": "off",
    "import/no-unresolved": "off",
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [".js", ".jsx", ".tsx", ".ts"]
      }
    ],
    "react/function-component-definition": [
      2,
      {
        "namedComponents": "arrow-function"
      }
    ],
    "semi": ["error", "always"],
    "indent": "off",
    "linebreak-style": ["error", "unix"],
    "quotes": ["error", "single"],
    "no-multi-spaces": "error",
    "@typescript-eslint/no-unused-vars": "warn",
    "space-in-parens": "error",
    "no-multiple-empty-lines": "error",
    "@typescript-eslint/no-explicit-any": "warn",
    "prettier/prettier": "error",
    "@typescript-eslint/consistent-type-imports": [
      "warn",
      {
        "prefer": "type-imports"
      }
    ],
    "import/order": [
      "warn",
      {
        "groups": [["builtin", "external", "internal"], "type"],
        "newlines-between": "always",
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": true
        }
      }
    ],
    "package-json/restrict-dependency-ranges": [
      "error",
      {
        "forDependencyTypes": ["devDependencies"],
        "rangeType": "pin"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"],
        "moduleDirectory": ["node_modules", "src"]
      },
      "typescript": {}
    }
  }
}
```

**Comment by michaelfaith** [comment]
Thanks for submitting again. I can see what's wrong.  In our v1 release, we dropped support for ESLint v8 and legacy rc configs.  So, it's not that the rule isn't outputting an error, it's that the plugin doesn't work at all with legacy configs.

Requirements listed in our Getting Started:

<img width="736" height="139" alt="screenshot" src="https://github.com/user-attachments/assets/b659ee37-5f18-4d09-9820-17c99730b5e9" />

Release Notes for v1: https://github.com/michaelfaith/eslint-plugin-package-json/releases/tag/v1.0.0

> - drop support for eslint v8 (https://github.com/michaelfaith/eslint-plugin-package-json/issues/1589)
> - drop legacy-recommended rc-based config (https://github.com/michaelfaith/eslint-plugin-package-json/issues/1579)

ESLint v8 went End of Life in October of 2024, a year and a half ago.  Looking at the config you shared, it should be pretty simple to migrate it to the new config format, and update ESLint to the latest major version.  What would be my recommendation.

**Comment by gadzhimari** [comment]
Thank you. Migration to ESLint 9 helped me.
